src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/esa/EsaWriteFinalDecisionPreviewDecisionService.java

```diff
@@ -53,43 +52,48 @@ protected void setTemplateContent(DecisionNoticeOutcomeService outcomeService, P
         NoticeIssuedTemplateBodyBuilder builder, SscsCaseData caseData,
         WriteFinalDecisionTemplateBody payload) {
 
-        // Validate here for ESA instead of only validating on submit.
-        // This ensures that we know we can obtain a valid allowed or refused condition below.
-        outcomeService.validate(response, caseData);
-
-        if (response.getErrors().isEmpty()) {
-
-            // If validation has produced no errors, we know that we can get an allowed/refused condition.
-            Optional<EsaAllowedOrRefusedCondition> condition = EsaPointsRegulationsAndSchedule3ActivitiesCondition
-                .getPassingAllowedOrRefusedCondition(decisionNoticeQuestionService, caseData);
-            if (condition.isPresent()) {
-                EsaScenario scenario = condition.get().getEsaScenario(caseData);
-                EsaTemplateContent templateContent = scenario.getContent(payload);
-                builder.writeFinalDecisionTemplateContent(templateContent);
-            } else {
-                // Should never happen.
-                response.addError("Unable to obtain a valid scenario - something has gone wrong");
+
+        if ("Yes".equalsIgnoreCase(caseData.getWriteFinalDecisionGenerateNotice())) {
+
+            // Validate here for ESA instead of only validating on submit.
+            // This ensures that we know we can obtain a valid allowed or refused condition below
+            outcomeService.validate(response, caseData);
+            if (response.getErrors().isEmpty()) {
+
+                // If validation has produced no errors, we know that we can get an allowed/refused condition.
+                Optional<EsaAllowedOrRefusedCondition> condition = EsaPointsRegulationsAndSchedule3ActivitiesCondition
+                    .getPassingAllowedOrRefusedCondition(decisionNoticeQuestionService, caseData);
+                if (condition.isPresent()) {
+                    EsaScenario scenario = condition.get().getEsaScenario(caseData);
+                    EsaTemplateContent templateContent = scenario.getContent(payload);
+                    builder.writeFinalDecisionTemplateContent(templateContent);
+                } else {
+                    // Should never happen.
```

Comment:
Do we want to log this message to help with debugging?

Reply:
Good idea - now logging